src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -379,6 +384,169 @@ public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationT
+            {
+                LogicalTimestamp kvsLogicalTimestamp = this.DeserializeLogicalTime(key, kvsValue);
+                LogicalTimestamp rcLogicalTimestamp = LogicalTimestampSerializer.Deserialize(rcValue);
+                expected = kvsLogicalTimestamp.ToString();
+                actual = rcLogicalTimestamp.ToString();
+                result = kvsLogicalTimestamp.Timestamp == rcLogicalTimestamp.Timestamp;
+            }
+            else if (key.StartsWith("Reminder"))
+            {
+                ActorReminderData kvsActorReminderData = this.DeserializeReminder(key, kvsValue);
+                ActorReminderData rcActorReminderData = ActorReminderDataSerializer.Deserialize(rcValue);
+                expected = kvsActorReminderData.ToString();
+                actual = rcActorReminderData.ToString();
+                result = kvsActorReminderData.ActorId == rcActorReminderData.ActorId
+                    && kvsActorReminderData.Name == rcActorReminderData.Name
+                    && kvsActorReminderData.DueTime == rcActorReminderData.DueTime
+                    && kvsActorReminderData.Period == rcActorReminderData.Period
+                    && kvsActorReminderData.State == rcActorReminderData.State
+                    && kvsActorReminderData.LogicalCreationTime == rcActorReminderData.LogicalCreationTime;
+            }
+            else
+            {
+                var message = "Migration Error: Failed to parse the KVS key - " + key;
+
+                ActorTrace.Source.WriteErrorWithId(
+                    this.TraceType,
+                    this.traceId,
+                    message);
+
+                var healthInfo = new HealthInformation(this.TraceType, message, HealthState.Error);
+                healthInfo.TimeToLive = TimeSpan.MaxValue;
+                healthInfo.RemoveWhenExpired = false;
+                this.servicePartition.ReportPartitionHealth(new HealthInformation(this.TraceType, message, HealthState.Error));
+            }
+
+            if (!result)
+            {
+                ActorTrace.Source.WriteErrorWithId(
+                    this.TraceType,
+                    this.traceId,
+                    $"Migrated data validation failed for key {key}. Expected Value: {expected} Actual Value: {actual}");
+            }
+
+            return result;
+        }
+
+        internal async Task AddOrUpdateMigratedKeysAsync(ITransaction tx, List<string> keysMigrated, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var keysMigratedCsv = string.Join(MigrationConstants.DefaultDelimiter.ToString(), keysMigrated);
+            ActorTrace.Source.WriteNoiseWithId(this.TraceType, this.traceId, "New keys migrated [{0}]", keysMigratedCsv);
+
+            var migrationKeysMigratedChunksCount = await this.metadataDictionary.TryGetValueAsync(tx, MigrationConstants.MigrationKeysMigratedChunksCount);
+            long chunksCount = 0L;
```

Comment:
Should the chunks count be 1 when you insert it for the first time?

Reply:
we are incrementing chunks count before adding to metadata dictionary. So it will be 1 when we insert it for the first time.